### PR TITLE
Hard deps and CSV->DelimitedFiles

### PR DESCRIPTION
**Pull request by tlienart**
This PR

* removes optional dependencies
* uses the stdlib `DelimitedFiles` to load `boston`, etc

See also https://github.com/alan-turing-institute/ScientificTypes.jl/pull/38; this addresses #77 ; ideally merge this in dev second (after merging https://github.com/alan-turing-institute/ScientificTypes.jl/pull/38)

### CODE REVIEW

File: src/datasets_requires.jl [hunk 1]
```diff
@@ -1,32 +1,172 @@
...
+#
+#   for (n, st) in zip(names(data), scitype_union.(eachcol(data)))
+#     println(":$n=>$st,")
+#   end
+#
+# 4. copy and paste the result in a coerce
+#
+#  data_table = coerce(data_table, ...)
+#
+# -------------------------------------------------------
```
Comment (ablaom) [inline src/datasets_requires.jl]: Thanks for the documentation here!

File: test/runtests.jl [hunk 2]
```diff
@@ -29,9 +29,9 @@ end
   @test include("info.jl")
 end
 
-@testset "datasets" begin
-  @test include("datasets.jl")
-end
+# @testset "datasets" begin
+#   @test include("datasets.jl")
+# end
```
Comment (ablaom) [inline test/runtests.jl]: Can we please have the tests back?

Comment (tlienart) [reply test/runtests.jl]: oops

**Comment by tlienart** [comment]
ok done, actually I'm glad you spotted it because the code for the datasets wasn't properly included  🙄  should be all good now

**Comment by ablaom** [comment]
It looks like the macro forms of loading data still import CSV:

```julia
macro load_boston()
    quote
        import CSV
        y, X = unpack(load_boston(), ==(:MedV), x->x != :Chas)
        (X, y)
    end
end
```

(As a side note: the raison d'etre for using macros - importing an optional dependency - is gone. Let's keep the macro - which has a different return value from the functions - because otherwise we break MLJ)

**Comment by tlienart** [comment]
thanks yes will remove those

File: src/datasets.jl [hunk 3]
```diff
@@ -1,9 +1,173 @@
...
+end
+
+
+load_boston()       = load_dataset("Boston.csv", COERCE_BOSTON)
+load_reduced_ames() = load_dataset("reduced_ames.csv", COERCE_REDUCED_AMES)
+load_ames()         = load_dataset("ames.csv", COERCE_AMES)
+load_iris()         = load_dataset("iris.csv", COERCE_IRIS)
+load_crabs()        = load_dataset("crabs.csv", COERCE_CRABS)
+
+load_crabs()
```
Comment (ablaom) [inline src/datasets.jl]: I'm not sure this fixes the travis fail, but line 166 is presumably not supposed to be here.

**Pull request merged**